### PR TITLE
Pure C implementation

### PR DESCRIPTION
**Pull request by aagrawal05**
Just got a the basic step030 hello triangle version working with a pure C implementation. Might be worth creating a new `step030-sdl-c` branch with the C code in case it is useful for other who are doing something similar (unlikely as that may be).

I've currently just opened this on the `step030-sdl` because I can't open new branches but I think, if you would like, you could open a new `step030-sdl-c` branch and change this pull request to target and merge with that instead.

The code could also be useful for creating the `step030-sdl-vanilla` mentioned in the appendix

### CODE REVIEW

**Comment by eliemichel** [comment]
Thank you for sharing this version! I am willing to merge it into a new `step030-sdl-c` branch and link to it in the guide, but before so I find a bit suspicious the `1738` files changed in the diff. ^^

Can you make sure you are using the very same version of the SDL as the current code does, as well as the same version of sdl2webgpu?

**Comment by aagrawal05** [comment]
Sure, will do.

On that note, I think it might be even easier to use a git submodules for all the external packages.

In my working directory I have the `sdl2webgpu`, `webgpu`, and `SDL2` as references to the appropriate branch of the repositories so it's not committed at all.

**Comment by aagrawal05** [comment]
Alright should be fixed

**Comment by eliemichel** [comment]
> On that note, I think it might be even easier to use a git submodules for all the external packages.

I find submodules a bit unfriendly as it is easy to forget clonning them. I'd rather use FetchContent like WebGPU-distribution does to get Dawn. GLFW is lightweight enough to be included as is (which I prefer because it greatly reduces the potential for error and breaking things) but indeed SDL is a bit too much for that.

**Comment by aagrawal05** [comment]
Sure, no problem. Should I also remove the unnecessary files in the webgpu distribution?

**Comment by aagrawal05** [comment]
I'll commit it just in case, you could just use the commit [36bac3a](https://github.com/eliemichel/LearnWebGPU-Code/commit/36bac3a5932a501b75dc9dc09beda8341137c3a2) if not needed.

**Comment by eliemichel** [comment]
Globally make sure that the diff between the C++ and the C versions is as minimal as it should be, so that it is easy to maintain (and portentially to batch cherry-pick commits when I want to bring the same change to many branches at once)

**Comment by aagrawal05** [comment]
Sure, I wouldn't mind making any changes necessary. 

Though I'm unsure exactly what you mean to change. The structure of both the C++ version and C version are identical. The only change is that I've 'translated' the requestAdapter and requestDevice from the 'webgpu.hpp' you wrote into the 'webpgu-utlils.h'.

File: sdl2webgpu/.github/workflows/cmake.yml
Comment (eliemichel) [inline sdl2webgpu/.github/workflows/cmake.yml:1]: No need for this file!

**Comment by eliemichel** [comment]
Just remove the workflow file from sdl2webgpu, and I thought I could re-route this PR to a different branch but apparently not, so I just created a `step030-sdl-c` branch, could you change the target of your PR? (Or re-open one if it is the only way?)

**Comment by aagrawal05** [comment]
Ah sorry, forgot about the .github folder!, sure will do. I'll just close this PR and open another one targeting the new branch.

**Comment by eliemichel** [comment]
(Replaced by #31)

**Pull request closed without merging**